doc/extensions/best-practices.rst

```diff
@@ -37,17 +39,61 @@ Implement each plugin class in a separate Python module
 This keeps CKAN's plugin loading order simple, see :ref:`ckan.plugins`.
 
 
-.. _extension config setting names best practice:
+.. _avoid name clashes:
+
+------------------
+Avoid name clashes
+------------------
+Many of the names you pick for your identifiers and files must be unique in
+relation to the names used by core CKAN and other extensions. To avoid
+conflicts you should prefix any public name that your extension introduces with
+the name of your extension. For example:
+
+* The names of *configuration settings* introduced by your extension should
+  have the form ``my_extension.my_config_setting``.
+
+* The names of *templates and template snippets* introduced by your extension
+  should begin with the name of your extension::
+
+      snippets/my_extension_useful_snippet.html
+
+  If you have add a lot of templates you can also put them into a separate
+  folder named after your extension instead.
+
+* The names of *template helper functions* introduced by your extension should
+  begin with the name of your extension. For example:
+
+  .. literalinclude:: /../ckanext/example_theme/v08_custom_helper_function/plugin.py
+     :pyobject: ExampleThemePlugin.get_helpers
+
+* The names of *JavaScript modules* introduced by your extension should begin
+  with the name of your extension. For example
+  ``fanstatic/example_theme_popover.js``:
+
+  .. literalinclude:: /../ckanext/example_theme/v16_initialize_a_javascript_module/fanstatic/example_theme_popover.js
+
+* The names of *API action functions* introduced by your extension should begin
+  with the name of your extension. For example
+  ``my_extension_foobarize_everything``.
+
+* The names of *background job queues* introduced by your extension should
+  begin with the name of your extension. For example
+  ``my_extension:super-special-job-queue``.
+
+In some situations, a resource may even be shared between multiple CKAN
+*instances*, which requires an even higher degree of uniqueness for the
+corresponding names. In that case, you should also prefix your identifiers with
+the CKAN site ID, which is available via
+
+::
 
------------------------------------------------------------------
-Names of config settings should include the name of the extension
------------------------------------------------------------------
+    from pylons import config
```

Comment:
See comment about `from ckan.common import config`. As this is only available on CKAN>=2.7 perhaps is best to show the import or show two examples, one for CKAN<2.7 and one for CKAN>=2.7